src/widgets/Notebook.cpp

```diff
@@ -1169,22 +1170,29 @@ SplitContainer *SplitNotebook::getOrAddSelectedPage()
 {
     auto *selectedPage = this->getSelectedPage();
 
-    return selectedPage != nullptr ? (SplitContainer *)selectedPage
-                                   : this->addPage();
+    if (selectedPage)
+    {
+        return (SplitContainer *)(selectedPage);
```

Comment:
```suggestion
        return static_cast<SplitContainer *>(selectedPage);
```

Reply:
We'd technically want this to be a dynamic cast, I'll look to see if messy it would be to chance the stored type from a `QWidget*` to a `SplitContainer*`

Reply:
Aren't we only ever storing `SplitContainer *`s anyway? (I think I recall that assumption being made in different places in the code as well.) If so, I believe  a `static_cast` would be correct

Reply:
It'll be quite a big change, basically merging the "Notebook" and "SplitNotebook" classes, so I'd prefer to defer this change to a refactor PR to keep this fix as small as possible. I only changed something here to make the code I was looking through readable

Reply:
nvm, Notebook is used in 3 places: 1. as a way to hold splitcontainers (the regular use). 2. controlling tabs in the "new channel" or "edit channel" popup. 3. the emote popup uses it. so the refactor would not merge the two, but rather look at what specializations make sense to keep in Notebook and what should be moved to SplitNotebook (which manages usecase 1.)